docs/CONTRIBUTING.md

```diff
@@ -12,6 +12,122 @@ Take a look at the [MongoDB connector](../connectors/sources/mongo.py) for more
 
 Each connector will have their own specific behaviors and implementations. When a connector is loaded, it stays in memory, so you can come up with any strategy you want to make it more efficient. You just need to be careful not to blow memory.
 
+### Sync rules
```

Comment:
General remark: this should be consistent with https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html. For example, I presume the introductory details are the same.

Reply:
I think we want to have a more technical description here as the description in the official docs is very ui-focused IMO.

Reply:
Definitely! But if there's anything useful here that isn't in the main docs, I think we should make a note to add it there too.

Reply:
Hmm, I'm not too sure about that. The main docs look like they're targeting users of connectors and the docs here target connector developers, which need some information, what's expected from them to implement and what not. I think we can copy this description, when we've a separate "How to build a connector" section in the main docs. 

WDYT?

Reply:
That sounds right 👍.